### PR TITLE
[POAE7-2465]Support BOOL for arrow format

### PR DESCRIPTION
**Pull request by imajiayu**
### What changes were proposed in this pull request?
Support BOOL for arrow format, add related function in ArrowArrayBuilder and QueryArrowArrayGenerator

### How was this patch tested?
add Unit Test in corresponding test files

### CODE REVIEW

File: cider/tests/utils/ArrowArrayBuilder.h [hunk 1]
```diff
@@ -129,6 +129,77 @@ class ArrowArrayBuilder {
...
+
+      size_t bitmap_size = (row_num_ + 7) >> 3;
+      void* null_buf = (void*)allocator_->allocate(bitmap_size);
+      std::memset(null_buf, 0xFF, bitmap_size);
+      for (auto i = 0; i < null_data.size(); i++) {
+        if (null_data[i]) {
+          CiderBitUtils::clearBitAt((uint8_t*)null_buf, i);
+        }
+      }
+      // TODO: null_count
```
Comment (yma11) [inline cider/tests/utils/ArrowArrayBuilder.h:179]: What's the details of this `TODO`? Can we fix it now? Or actually we don't use `null_count` for current computation?

File: cider/tests/utils/ArrowArrayBuilderTest.cpp [hunk 2]
```diff
@@ -94,6 +94,20 @@ TEST(ArrowArrayBuilderTest, RowNum) {
...
+  ArrowArray* array = nullptr;
+  ArrowSchema* schema = nullptr;
+
+  std::vector<bool> vec{true, false, true, false, true, true, false, false};
+  std::tie(schema, array) =
+      ArrowArrayBuilder().setRowNum(8).addBoolColumn<bool>("bool_list", vec).build();
+
+  EXPECT_EQ(8, array->length);
+  EXPECT_EQ(1, array->n_children);
+  EXPECT_EQ(0x35, *(uint8_t*)(array->children[0]->buffers[1]));
```
Comment (jikunshang) [inline cider/tests/utils/ArrowArrayBuilderTest.cpp:107]: `0x35` -> `0b00110101`

File: cider/tests/utils/ArrowArrayBuilderTest.cpp [hunk 3]
```diff
@@ -94,6 +94,20 @@ TEST(ArrowArrayBuilderTest, RowNum) {
       CiderCompileException);
 }
 
+TEST(ArrowArrayBuilderTest, BoolColumnBatch) {
```
Comment (jikunshang) [inline cider/tests/utils/ArrowArrayBuilderTest.cpp:97]: I suggest to add more boundary condition like bit length cannot align to 8. (maybe we can add more when checker part ready)

**Review by jikunshang — APPROVED** [review]

**Review by yma11 — APPROVED** [review]
For the minor comments, let's address it in later PR and merge it first.
